src/core/varinfo.jl

```diff
@@ -2,21 +2,28 @@ import Base.string, Base.isequal, Base.==, Base.convert
 import Base.getindex, Base.setindex!
 import Base.rand, Base.show
 
-########## VarName ##########
-
+###########
+# VarName #
+###########
 immutable VarName
   csym      ::    Symbol        # symbol generated in compilation time
   sym       ::    Symbol        # variable symbol
   indexing  ::    String        # indexing
   counter   ::    Int           # counter of same {csym, uid}
 end
 
+typealias UID Tuple{Symbol,Symbol,String,Int}
```

Comment:
I meet further problem of this point. I found the problem that we cannot use `VarName` as a key origins from the way how Julia treats strings inside types. A minimum example is given below.

```julia
immutable B
    s   ::  String
end
b1 = B("1")
b2 = B("2")
hash(b1.s) # => 0x8f6e84dd949e9002
hash(b2.s) # => 0x2ba9e56d0c27e607
b1.s == b2.s # => false
```

This means no matter how I define hash it won't work because the strings inside are "different". Do you know why this happens?